cdb2api/cdb2api.c

```diff
@@ -3626,6 +3639,24 @@ int cdb2_get_effects(cdb2_hndl_tp *hndl, cdb2_effects_tp *effects)
     return rc;
 }
 
+static void free_query_list(struct query_list *head)
+{
+    struct cdb2_query *q, *tmp;
+    TAILQ_FOREACH_SAFE(q, head, entry, tmp)
+    {
+        TAILQ_REMOVE(head, q, entry);
+        free(q->sql);
+        free(q->buf);
+        free(q);
+    }
+}
+
+/* Free query list and reset the pointer. */
+static inline void free_query_list_on_handle(cdb2_hndl_tp *hndl)
+{
+    free_query_list(&hndl->queries);
```

Comment:
should we `TAILQ_INIT` here?

Reply:
Also bbcdb2api doesn't call tailq_init in its free function